### PR TITLE
Add CRD Clientset, Lister Informers

### PR DESCRIPTION
**Pull request by christopherhein**
*Issue #, if available:* #226 

*Description of changes:*
* This changes the `pkg/apis/crd` to use the standard `pkg/apis/{groupName}` path.
* Adds new shell script for generating the `pkg/client` libs using the `code-generator` project
* Adds the client packages 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/update-codegen.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+set -o errexit
+set -o nounset
+set -o pipefail
+ SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
+ # generate the code with:
```
Comment (mogren) [inline scripts/update-codegen.sh]: Nit: Some extra whitespaces here.

Comment (christopherhein) [reply scripts/update-codegen.sh]: Thanks for pointing that out, fixed.

**Comment by christopherhein** [comment]
@mogren thanks for that feedback, updated the `update-codegen.sh` file

**Comment by christopherhein** [comment]
@mogren @nckturner fixed conflicts with the ipamd tests w/ `master`.

**Comment by mogren** [comment]
@christopherhein What is the plan around this?

**Comment by christopherhein** [comment]
@mogren this should be still necessary. In essence this would allow someone to be able to build a controller that could use the `ENIConfig` CRDs without forcing them to use OperatorSDK to load the resources like we do in this project. You'd be able to use the `informer` and `clientsets` similar to what is used in https://github.com/kubernetes/sample-controller 

The code is nearly 100% code generated except for a couple fixes to how the resources are organized and making sure to define in the `types.go` that they should be `nonNamespaced` ie `Cluster` scoped.

**Comment by christopherhein** [comment]
FYI @mogren just rebased this since it had conflicts. Let me know how you want to handle moving this forward.

**Comment by jaypipes** [comment]
@christopherhein yo! OK if either myself or @mogren take this over and rebase up to master? Or are you still wanting to work on this?

**Comment by christopherhein** [comment]
Please do, I don’t have this use case anymore.

**Comment by jaypipes** [comment]
> Please do, I don’t have this use case anymore.

10-4 good buddy.

**Pull request closed without merging**